Userland/Libraries/LibDebug/Dwarf/CompilationUnit.h

```diff
@@ -15,8 +15,6 @@
 namespace Debug::Dwarf {
 
 class DwarfInfo;
-class DIE;
-class LineProgram;
```

Comment:
It can't, because CompilationUnit contains
```
NonnullOwnPtr<LineProgram> m_line_program;
```
and an implicit destructor, which needs to be able to call the destructor of LineProgram. The latter is impossible with an incomplete definition of LineProgram.

Reply:
Ah, the common workaround for that is to add an out-of-line `= default` destructor in the .cpp file (which IMO is preferable over not being able to forward declare)

Reply:
I can do that, sure, but that sounds like extra pain, and also `line_program()` must move out of line. What for? Breaking the cycle at LineProgram is sufficient, it doesn't need to be broken again.

Reply:
Reducing `#include`s helps build times (a *lot*, over a whole codebase) and prevents further opportunity for cycles. I'd argue that's less painful than writing `CompilationUnit::CompilationUnit() = default` in a .cpp file.

Anyways, it's more of a "while we're here doing the right thing, let's do more of the right thing" point, but I'm not gonna hold up this PR for it.

Reply:
Yeah, good point, you're right. I'll add it. :+1: